service/src/main.rs

```diff
@@ -546,11 +538,65 @@ impl InclusionService {
         };
     }
 
+    /// Helper function to handle error from a SP1 NetworkProver Clents.
+    /// Will finalize the job in an [JobStatus::Failed] state,
+    /// that may be retryable.
+    fn handle_zk_client_error(
+        &self,
+        zk_client_error: &SP1NetworkError,
+        job: &Job,
+        job_key: &Vec<u8>,
+    ) -> InclusionServiceError {
+        error!("SP1 Client error: {zk_client_error}");
+        let (e, job_status);
+        match zk_client_error {
+            SP1NetworkError::SimulationFailed | SP1NetworkError::RequestUnexecutable { .. } => {
+                e = InclusionServiceError::DaClientError(
+                    format!("ZKP program critical failure: {zk_client_error} occured for {job:?} PLEASE REPORT!"),
+                );
+                job_status = JobStatus::Failed(e.clone(), None);
+            }
+            SP1NetworkError::RequestUnfulfillable { .. } => {
+                e = InclusionServiceError::DaClientError(format!(
+                    "ZKP network failure: {zk_client_error} occured for {job:?} PLEASE REPORT!"
+                ));
+                job_status = JobStatus::Failed(e.clone(), None);
+            }
+            SP1NetworkError::RequestTimedOut { request_id } => {
+                e = InclusionServiceError::DaClientError(format!(
+                    "ZKP network: {zk_client_error} occured for {job:?}"
+                ));
+
+                let id = request_id
+                    .as_slice()
+                    .try_into()
+                    .expect("request ID is always correct length");
+                job_status =
+                    JobStatus::Failed(e.clone(), Some(JobStatus::ZkProofPending(id).into()));
+            }
+            SP1NetworkError::RpcError(_) | SP1NetworkError::Other(_) => {
+                e = InclusionServiceError::DaClientError(format!(
+                    "ZKP network failure: {zk_client_error} occured for {job:?} PLEASE REPORT!"
+                ));
+                // TODO: We cannot clone KeccakInclusionToDataRootProofInput thus we cannot insert into a JobStatus::DataAvalibile(proof_input)
+                // So we just redo the work from scratch for the DA side as a stupid workaround
+                job_status =
+                    JobStatus::Failed(e.clone(), Some(JobStatus::DataAvailabilityPending.into()));
```

Comment:
Not great. Interally to that type,there is a `hasher` that isn't Copy/Clone we likely can't get (at least directly.

Thinking of other ways that don't need a clone here :thinking: